Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptDeleteController.java

```diff
@@ -1,163 +1,175 @@
+     *            A generic model holder for Servlet
+     * @return String value to redirect user to a particular concept list page
+     * @throws IndexerRunningException 
+     */
+    @RequestMapping(value = "auth/conceptlist/deleteconceptconfirm/{id}", method = RequestMethod.GET)
+    public ModelAndView confirmlDelete(@PathVariable("id") String id,
+            @RequestParam(value = "fromHomeScreenDelete") String fromHomeScreenDelete,Principal principal)
+                    throws LuceneException, IndexerRunningException {
+        List<ConceptEntryWrapper> foundConcepts = null;
+        ModelAndView model = new ModelAndView();
+        ConceptEntry concept = conceptManager.getConceptEntry(id);
+        ChangeEvent changeEvent = new ChangeEvent();
+            changeEvent.setType(ChangeEventConstants.DELETION);
+            changeEvent.setDate(new Date());
+            changeEvent.setUserName(principal.getName());
+            List<ChangeEvent> ChangeEventsList = new ArrayList<ChangeEvent>();
+            ChangeEventsList.add(changeEvent);
+            concept.setChangeEvents(ChangeEventsList);
+        //Check if indexer is running
+        if (indexService.isIndexerRunning()) {
+            model.addObject("show_error_alert", true);
+            model.addObject("error_alert_msg", indexerRunning);
+            //Need to include command Object
+            model.setViewName("/auth/conceptlist/deleteconcept");
+            return model;
+        }
+        
+        conceptManager.deleteConcept(id, principal.getName());
+        List<ConceptEntry> founds = conceptManager.getConceptListEntries(concept.getConceptList());
+
+        foundConcepts = wrapperCreator
+                .createWrappers(founds != null ? founds.toArray(new ConceptEntry[founds.size()]) : new ConceptEntry[0]);
+
+        model.addObject("result", foundConcepts);
+        if (fromHomeScreenDelete.equalsIgnoreCase("true")) {
+            model.setViewName("redirect:/login");
+            return model;
+        }
+        model.setViewName("/auth/conceptlist/concepts");
+        return model;
+    }
+
+    @RequestMapping(value = "auth/conceptlist/deleteconcepts/{id}", method = RequestMethod.GET)
+    public ModelAndView deleteConcept(@PathVariable("id") String id,
+            @ModelAttribute("conceptSearchBean") ConceptSearchBean conceptSearchBean, BindingResult result, Principal principal)
+                    throws LuceneException, IndexerRunningException {
+        ModelAndView model = new ModelAndView();
+        // Check if indexer is running
+        if (indexService.isIndexerRunning()) {
+            model.addObject("show_error_alert", true);
+            model.addObject("error_alert_msg", indexerRunning);
+            // Need to include command Object
+            model.setViewName("welcome");
+            return model;
+        }
+        conceptManager.deleteConcept(id, principal.getName());
+        model.setViewName("welome");
+        return model;
+    }
+
```

Comment:
@jdamerow  I have added username field to conceptManager.deleteConcept() method.